src/core/types/secret_bytes.rs

```diff
@@ -11,26 +11,74 @@ macro_rules! decl_secret_bytes {
     ) => {
         paste! {
 
+            #[derive(Zeroize, ZeroizeOnDrop, derive_more::Debug, derive_more::Display)]
+            #[debug("OBFUSCATED")]
+            #[display("OBFUSCATED")]
+            pub struct [< $struct_name SecretMagic >](Box<[u8; $byte_count]>);
+
+            uniffi::custom_type!([< $struct_name SecretMagic >], BagOfBytes);
+
+            impl TryFrom<BagOfBytes> for [< $struct_name SecretMagic >] {
+                type Error = CommonError;
+                fn try_from(value: BagOfBytes) -> Result<Self> {
+                    let fixed_size: &[u8; $byte_count] = value.as_ref().try_into().map_err(|_| CommonError::InvalidByteCount { expected: $byte_count as u64, found: value.len() as u64 })?;
+                    Ok(Self(Box::new(*fixed_size)))
+                }
+            }
+
+            impl $crate::UniffiCustomTypeConverter for [< $struct_name SecretMagic >] {
+                type Builtin = BagOfBytes;
+
+                fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+                    Self::try_from(val).map_err(|e| e.into())
+                }
+
+                fn from_custom(obj: Self) -> Self::Builtin {
+                    BagOfBytes::from(obj.0.as_slice())
+                }
+            }
+
             $(
                 #[doc = $expr]
             )*
-            #[derive(Zeroize, ZeroizeOnDrop, derive_more::Debug, derive_more::Display)]
+            #[derive(Zeroize, derive_more::Debug, derive_more::Display, uniffi::Record)]
             #[debug("OBFUSCATED")]
             #[display("OBFUSCATED")]
-            pub struct $struct_name(Box<[u8; Self::LENGTH]>);
+            pub struct $struct_name {
+                secret_magic: [< $struct_name SecretMagic >]
```

Comment:
using the `secret_magic` trick as usual, making the `$struct_name` its own concrete type, rather than a typealias, on the FFI side.